tuist/tuist
# Add knownRegions to generated projects

Part of: #659 

### Short description 📝

Generated projects were missing an exhaustive list of known regions. This causes Xcode to immediately modify generated projects on launch to fix this. 

### Solution 📦

The list of known regions can be included in the generated projects. Based on localizable resources we can infer the known regions of the project.

This can be achieved by keeping track of the localizations encountered when adding while adding localizable resource file elements.

### Implementation 👩‍💻👨‍💻

- [x] Update `ProjectFileElements` to keep track of localizations
- [x] Update `ProjectGenerator` to set the appropriate `knownRegions` based on the localisations tracked by the `ProjectFileElements`

### Test Plan 🛠

- Run `tuist generate` within `fixtures/ios_app_with_framework_and_resources`
- Verify the generated project MainApp has both "English" and "French" in the list of localizations in the project's info tab
